### PR TITLE
Support multiple chained lookups per position

### PR DESCRIPTION
**Pull request by simoncozens**
See  #1119. This allows for e.g. `sub a' b lookup l1 lookup l2;` ~~I am not quite sure if it works for pos rules too, although that's probably a one-character fix in the grammar if not.~~ and `pos a' lookup l1 lookup l2;`

See also independent Python implementation in fontTools/fontTools#1905.

### CODE REVIEW

**Comment by cjchapman** [comment]
@simoncozens Please rebase this after PR #1127 has been reviewed and merged. Thanks!

@punchcutter Please wait until Simon's rebased this before reviewing. Thanks!

File: c/makeotf/makeotf_lib/build/hotpccts/featgram.g [hunk 1]
```diff
@@ -725,11 +725,15 @@ pattern[int markedOK]>[GNode *pat]
 						zzerr("Glyph or glyph class must precede a lookup reference in a contextual rule.");
 						
 					labelIndex = featGetLabelIndex($t.text);
-					(*insert)->lookupLabel = labelIndex;
+					(*insert)->lookupLabels[(*insert)->lookupLabelCount] = labelIndex;
+					(*insert)->lookupLabelCount++;
+					if ((*insert)->lookupLabelCount > 255)
+						zzerr("Too many lookup references in one glyph position.");
+					printf("Lookup label count is %i, index is %i\n", (*insert)->lookupLabelCount, (*insert)->lookupLabels[(*insert)->lookupLabelCount-1]);
```
Comment (punchcutter) [inline c/makeotf/makeotf_lib/build/hotpccts/featgram.g:732]: Can we drop the printf?

File: c/makeotf/makeotf_lib/source/hotconv/featgram.c [hunk 2]
```diff
@@ -1172,24 +1172,23 @@ pattern(int markedOK)
                                         zzerr("Glyph or glyph class must precede a lookup reference in a contextual rule.");
 
                                     labelIndex = featGetLabelIndex(t.text);
-                                    (*insert)->lookupLabel = labelIndex;
+                                    (*insert)->lookupLabels[(*insert)->lookupLabelCount] = labelIndex;
+                                    (*insert)->lookupLabelCount++;
+                                    if ((*insert)->lookupLabelCount > 255)
+                                        zzerr("Too many lookup references in one glyph position.");
+                                    printf("Lookup label count is %i, index is %i\n", (*insert)->lookupLabelCount, (*insert)->lookupLabels[(*insert)->lookupLabelCount - 1]);
```
Comment (punchcutter) [inline c/makeotf/makeotf_lib/source/hotconv/featgram.c:1179]: Same here. My terminal is getting ugly with all this printing.

File: c/makeotf/makeotf_lib/source/hotconv/feattoks.h [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #ifndef feattoks_h
 #define feattoks_h
-/* feattoks.h -- List of labeled tokens and stuff
+/* feattoks.h -- List of labelled tokens and stuff
```
Comment (punchcutter) [inline c/makeotf/makeotf_lib/source/hotconv/feattoks.h:3]: Well, ok, but the American spelling is also fine.

File: docs/OpenTypeFeatureFileSpecification.md [hunk 4]
```diff
@@ -1648,11 +1648,11 @@ sequence are written in reverse order from the text to be matched. Developers of
 font editing tools who know this are sometimes confused by the FEA syntax.
 
 For each glyph or glyph class in the input sequence, the contextual rule may
-specify one lookup (§[4.e](#4.e)) to be applied at that position. Note that the
-specified lookup may contain many rules; the implementation must ensure that
-only one rule in a referenced lookup will match at that position in the input
-sequence. Lookups cannot be specified for the glyphs or glyph classes in the
-backtrack and lookahead sequences.
+specify one or more lookups (§[4.e](#4.e)) to be applied at that position.
+Note that the specified lookup may contain many rules; the implementation
```
Comment (punchcutter) [inline docs/OpenTypeFeatureFileSpecification.md:1652]: Should this say "the specified lookups" plural?

File: docs/OpenTypeFeatureFileSpecification.md [hunk 5]
```diff
@@ -2280,11 +2280,11 @@ sequence are written in reverse order from the text to be matched. Developers of
 font editing tools who know this are sometimes confused by the FEA syntax.
 
 For each glyph or glyph class in the input sequence, the contextual rule may
-specify one lookup (§[4.e](#4.e)) to be applied at that position. Note that the
-specified lookup may contain many rules; the implementation must ensure that
-only one rule in a referenced lookup will match at that position in the input
-sequence. Lookups cannot be specified for the glyphs or glyph classes in the
-backtrack and lookahead sequences.
+specify one or more lookups (§[4.e](#4.e)) to be applied at that position. Note
+that the specified lookup may contain many rules; the implementation must
```
Comment (punchcutter) [inline docs/OpenTypeFeatureFileSpecification.md:2284]: Same as above. Should this say "the specified lookups" plural?

File: c/makeotf/makeotf_lib/source/hotconv/feattoks.h (hunk 3 above)
Comment (simoncozens) [reply c/makeotf/makeotf_lib/source/hotconv/feattoks.h:3]: Huh. I think that was antlr’s i18n, not me, but I’ll drop it from the commit.

File: docs/OpenTypeFeatureFileSpecification.md (hunk 4 above)
Comment (simoncozens) [reply docs/OpenTypeFeatureFileSpecification.md:1652]: Yep.

**Comment by simoncozens** [comment]
OK, that's all in. Waiting for Khaled's PR to drop then I'll rebase.

**Comment by punchcutter** [comment]
Awesome, I've been wanting to get this in for a while. Thanks for getting it done.

**Comment by simoncozens** [comment]
Done, I think, assuming I haven't messed up the git-fu.

**Pull request merged**